src/du_parameters/frequency.py

```diff
@@ -0,0 +1,526 @@
+        """Return the ARFCN as a string for development purposes."""
+        return f"ARFCN(channel={self._channel})"
+
+    def __str__(self) -> str:
+        """Return the ARFCN as a string."""
+        return str(self._channel)
+
+    def __add__(self, other: "ARFCN | int | Decimal") -> "ARFCN":
+        """Add another ARFCN or integer to this ARFCN.
+
+        Args:
+            other: The value to add (ARFCN or int or Decimal).
+
+        Returns:
+            ARFCN: A new ARFCN instance with the updated channel.
+
+        Raises:
+            NotImplementedError: If the other value is not an ARFCN or int.
+        """
+        if isinstance(other, ARFCN):
+            return ARFCN(self._channel + other._channel)  # type: ignore[operator]
+        if isinstance(other, int | Decimal):
+            return ARFCN(self._channel + other)
+        raise NotImplementedError(f"Unsupported type for addition: {type(other).__name__}")
+
+    def __eq__(self, other) -> bool:
+        """Check if ARFCN instance and other are equal.
+
+        Args:
+            other (ARFCN | Decimal | int): The value to compare with.
+
+        Returns:
+            ARFCN: A new ARFCN instance with the updated channel.
+        """
+        if isinstance(other, int | Decimal):
+            return other == self._channel
+        if isinstance(other, ARFCN):
+            return other._channel == self._channel
+        raise NotImplementedError(f"Unsupported type for equality: {type(other).__name__}")
+
+    @classmethod
+    def from_frequency(cls, frequency: Frequency) -> "ARFCN":
+        """Find the closest ARFCN corresponding to a given frequency.
+
+        Args:
+            frequency (Frequency): The input frequency instance.
+
+        Returns:
+            ARFCN (ARFCN): The closest ARFCN value to the given frequency.
+
+        Raises:
+            ValueError: If no configuration is found for the given frequency
+            or frequency is out of range.
+        """
+        config = get_range_from_frequency(frequency)
+        if config is None:
+            raise ValueError(f"No configuration found for frequency {frequency}")
+        offset = (frequency - config.freq_offset) / config.freq_grid
+        result = config.arfcn_offset + round(Decimal(offset))
+        return result
+
+
+class GSCN:
+    """Represent a Global Synchronization Channel Number (GSCN) used in 5G.
+
+    Include conversions to frequencies and validity checks.
+
+    Args:
+        channel (GSCN): The input GSCN.
+
+    Raises:
+        ValueError: If the GSCN is not within the valid range for GSCN or n is out of range.
+        NotImplementedError: If the other value is not an GSCN, int or Decimal.
+    """
+
+    def __init__(self, channel: "GSCN | int | Decimal"):
+        if channel < 0 or channel > 26639:  # type: ignore[operator]
+            raise ValueError(f"GSCN must be between 0 and 26639, got {channel} instead.")
+        self._channel = channel
```

Comment:
Right now we can accept a Decimal that represents a value that is not an integer.